src/Avalonia.Controls.TreeDataGrid/Selection/TreeDataGridRowSelectionModel.cs

```diff
@@ -88,6 +88,15 @@ void ITreeDataGridSelectionInteraction.OnKeyDown(TreeDataGrid sender, KeyEventAr
                     {
                         e.Handled = MoveSelection(sender, direction.Value, shift, anchor);
                     }
+
+                    if (!e.Handled && direction == NavigationDirection.Left
+                        && anchor?.Rows is HierarchicalRows<T> hierarchicalRows && anchorRowIndex > 0)
+                    {
+                        var newIndex = hierarchicalRows.GetParentRowIndex(AnchorIndex);
+                        UpdateSelection(sender, newIndex, true);
+                        sender.RowsPresenter.BringIntoView(newIndex);
+                        FocusManager.Instance.Focus(sender);
```

Comment:
This row is needed to fix the focus loss problem. No idea why it gets lost though